expr/builtins/time.go

```diff
@@ -528,18 +529,21 @@ func timeTruncEvalOne(ctx expr.EvalContext, args []value.Value) (value.Value, bo
 }
 func timeTruncEvalTwo(ctx expr.EvalContext, args []value.Value) (value.Value, bool) {
 
-	valTs := int64(0)
+	milliTs := int64(0)
+	microTs := int64(0)
 	switch itemT := args[0].(type) {
 	case value.TimeValue:
-		// Get the full Unix timestamp w/ milliseconds.
-		valTs = itemT.Int()
+		// Get full Unix timestamps w/ microseconds and milliseconds.
+		microTs = itemT.Time().In(time.UTC).UnixNano() / 1e3
+		milliTs = itemT.Int()
 	default:
 		// If not a TimeValue, convert to a TimeValue and get Unix w/ milliseconds.
 		t, ok := value.ValueToTime(args[0])
 		if !ok || t.IsZero() {
 			return value.NewStringValue(""), false
 		}
-		valTs = t.UnixNano() / 1e6
+		microTs = t.UnixNano() / 1e3
+		milliTs = microTs / 1e3
```

Comment:
I see what you are doing here, but I think it's a little confusing. Might just be me though

Reply:
yeah, do you think deriving both from `t.UnixNano()` would be clearer?

Reply:
I think my confusion was mainly looking at the diff and seeing you used a different number. I think its fine